### PR TITLE
fix(i): Index directive field arg name

### LINKED ISSUE #3132: Index directive `name` argument should be `field`
Current form: `@index(includes: [{name: "score", direction: DESC}])`

Should be: `@index(includes: [{field: "score", direction: DESC}])`

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #3132

## Description

This PR fixes a mistake I made when refactoring the index directive. The `name` argument of the `includes` fields has been changed to `field`.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Updated unit and integration tests.

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: internal/request/graphql/schema/types/types.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ const (
 	IndexDirectivePropDirection = "direction"
 	IndexDirectivePropIncludes  = "includes"
 
-	IndexFieldInputName      = "name"
+	IndexFieldInputName      = "field"
```
Comment (fredcarle) [inline internal/request/graphql/schema/types/types.go:42]: question: `IndexFieldInputName` should be `InputFieldInputField` to match the other contants naming convention. You preferred keeping it as `IndexFieldInputName`, can you explain why?

**Review by fredcarle — APPROVED** [review]
LGTM! `field` is definitely more appropriate and even less confusing.

Just one question.

File: internal/request/graphql/schema/types/types.go (hunk 1 above)
Comment (nasdf) [reply internal/request/graphql/schema/types/types.go:42]: It seemed more confusing to me if changed.

Comment (fredcarle) [reply internal/request/graphql/schema/types/types.go:42]: How about 
```go
IndexDirectiveFieldLabel = "field"
IndexDirectiveFieldDirection = "direction"
```
or maybe just
```go
IncludesPropField = field
IncludesPropDirection = direction
```

Comment (nasdf) [reply internal/request/graphql/schema/types/types.go]: Updated to your second suggestion

**Review by shahzadlone — APPROVED** [review]
Like the change! 👍

**Pull request merged**